### DISCUSSION TITLE
Trying to change clangd's c++ standard.

### DISCUSSION THREAD

**Discussion by mikebarbon** [helix-editor/helix#7626] [Q&A]
>  std::string toString() {
>     std::string stringValue = std::visit(
>         [](const auto &value) { return std::to_string(value); }, this->literal);
>   }

I get "no matching function for call to 'visit'". I understand that this is because clangd is using a standard below c++17.

In my project directory I have a compile_flags.txt:

`-std=c++20
-Wall
-I/usr/lib/gcc/x86_64-redhat-linux/13/include
-L/usr/lib/gcc/x86_64-redhat-linux/13/
`
This is my languages.toml:

> [[language]]
> name = "cpp"
> scope = "source.cpp"
> injection-regex = "cpp"
> file-types = ["cc", "hh", "c++", "cpp", "hpp", "h", "ipp", "tpp", "cxx", "hxx", "ixx", "txx", "ino", "C", "H", "cu", "cuh", "cppm"]
> roots = []
> comment-token = "//"
> indent = { tab-width = 2, unit = "  " }
> auto-format = true
> config = {"clangd.fallbackFlags" = [ "-std=c++20" ] }

What am I doing wrong?

**Comment by h1t** [comment]
Did you try this snippet?

```
[language-server.clangd]
config = { "clangd.fallbackFlags" = [ "-std=c++20" ] }
auto-format = true
```

**Reply by sgj123456** [reply]
I tried, but it still doesn't work

**Reply by Lisible** [reply]
I'm having the same problem here, trying to set the standard to c2x but clangd still outputs warnings related to -Wdeclaration-after-statement, even though this shouldn't output any warning using any standard >=c99.

EDIT: Might be a clangd issue, I'm reproducing it on VSCodium with clangd extension. I will investigate further if I have some time.

**Reply by sgj123456** [reply]
> I'm having the same problem here, trying to set the standard to c2x but clangd still outputs warnings related to -Wdeclaration-after-statement, even though this shouldn't output any warning using any standard >=c99.
> 
> EDIT: Might be a clangd issue, I'm reproducing it on VSCodium with clangd extension. I will investigate further if I have some time.

I found that using cmake is recommended in wikis, and it works normally after I use it
